src/cow-react/modules/limitOrders/hooks/useFetchInitialPrice.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState, useCallback } from 'react'
+import { useAtomValue } from 'jotai'
+import { useWeb3React } from '@web3-react/core'
+
+import { getNativePrice } from '@cow/api/gnosisProtocol/api'
+import { WrappedTokenInfo } from 'state/lists/wrappedTokenInfo'
+import { Field } from 'state/swap/actions'
+import { limitRateAtom } from '../state/limitRateAtom'
+import { useLimitOrdersTradeState } from './useLimitOrdersTradeState'
+import { limitDecimals } from '../utils/limitDecimals'
+
+function _getAddress(currency: WrappedTokenInfo): string | null {
+  let address: string | null = null
+
+  if (!currency) {
+    return null
+  }
+
+  if (currency.address) {
+    address = currency.address
+  }
+
+  if (currency.tokenInfo) {
+    address = currency.tokenInfo.address
+  }
+
+  return address
+}
+
+// Fetches the INPUT and OUTPUT price and calculates initial Active rate
+export function useFetchInitialPrice() {
```

Comment:
1. Please, set the return type
2. The hook returns value, so better give it `get` prefix instead of `fetch`

```
export function useGetLimitOrderMarketRate(): { price: string | null } {
```